File: src/main/java/com/azilzor/chatterchain/controller/UserInterfaceController.java
package com.azilzor.chatterchain.controller;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

import com.azilzor.chatterchain.security.CustomUserDetails;

@Controller
public class UserInterfaceController {

  @GetMapping("/")
  public String index(Model model, @AuthenticationPrincipal CustomUserDetails user) {
    if (user == null) {
      return "redirect:login";
    }

    model.addAttribute("nickname", user.getUsername());
    return "index";
  }

  @GetMapping("/login")
  public String login() {
    return "login";
  }

}

File: src/main/kotlin/com/azilzor/chatterchain/controller/UserInterfaceController.kt
package com.azilzor.chatterchain.controller

import com.azilzor.chatterchain.security.CustomUserDetails
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping

@Controller
class UserInterfaceController {
    @GetMapping("/")
    fun index(model: Model, @AuthenticationPrincipal user: CustomUserDetails?): String {
        if (user == null) {
            return "redirect:login"
        }

        model.addAttribute("nickname", user.username)
        return "index"
    }

    @GetMapping("/login")
    fun login(): String {
        return "login"
    }
}
